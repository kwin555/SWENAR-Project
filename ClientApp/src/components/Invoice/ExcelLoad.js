import React, { Component } from "react";

class ExcelLoad extends Component {
  static displayName = ExcelLoad.name;

  constructor(props) {
    super(props);
    this.state = {};
  }
  // component used to Post an excel file to the database
  render() {
    return (
      <div>
        <form
          enctype='multipart/form-data'
          method='post'
          action='https://localhost:44375/api/Invoice/Load'
        >
          <input
            type='file'
            name='excelFile'
            id='excel-file'
            className='form-control'
          />
          <button type='submit'>Submit</button>
        </form>
      </div>
    );
  }
}

export default ExcelLoad;
