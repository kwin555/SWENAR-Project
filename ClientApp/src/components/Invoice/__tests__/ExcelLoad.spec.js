import React from "react";
import ExcelLoad from "../ExcelLoad";
import { mount } from "enzyme";

describe("ExcelLoad tests", () => {
  it("Excel load renders corrrectly", () => {
    const props = {};
    const wrapper = mount(<ExcelLoad {...props}></ExcelLoad>);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
