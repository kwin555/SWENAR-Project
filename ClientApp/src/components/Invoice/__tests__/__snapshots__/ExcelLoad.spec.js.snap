// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExcelLoad tests Excel load renders corrrectly 1`] = `
<ExcelLoad>
  <div>
    <form
      action="https://localhost:44375/api/Invoice/Load"
      enctype="multipart/form-data"
      method="post"
    >
      <input
        className="form-control"
        id="excel-file"
        name="excelFile"
        type="file"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</ExcelLoad>
`;
